src/main/java/core/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,37 @@
+package core.dao.impl;
+
+import core.dao.OrderDao;
+import core.lib.Dao;
+import core.model.Order;
+import core.model.User;
+import core.model.exception.DataProcessingException;
+import core.util.HibernateUtils;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl extends AbstractDao<Order> implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        return super.create(order, HibernateUtils.getSessionFactory());
+    }
+    
+    @Override
+    public List<Order> getOrdersHistory(User user) {
+        try (Session session = HibernateUtils.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery("from Order o "
+                                                     + "left join fetch o.tickets t "
```

Comment:
please see this conversation
https://mateacademy.slack.com/archives/G01EHB8CWAE/p1612455330236200

Reply:
Corrected